### PR TITLE
[Themes] Add support for block liquid files in theme partitioning

### PR DESCRIPTION
**Pull request by jamesmengo**
### WHY are these changes introduced?

Closes https://github.com/Shopify/develop-advanced-edits/issues/468

Adds support for theme block files when uploading theme

### WHAT is this pull request doing?
- Introduces explicit partitioning support for blocks folder
- Modifies theme uploads to upload blocks first

### How to test your changes?

1. Try uploading `horizon` from the CLI on the `CLI 3.71`+
2. Run `shopify theme push -u`
3. Verify that `custom-section` has an error like this: `Failed to upload file sections/custom-section.liquid:
-Invalid preset "Featured collection": invalid block type "featured-collection": undefined block type.`
4. Check out this branch and build
5. Run `pnpm shopify theme push -u`
6. Confirm that theme is uploaded with no errors

BEFORE

AFTER

### Measuring impact

- [x] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix

### Checklist

- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [x] I've considered possible [documentation](https://shopify.dev) changes

### CODE REVIEW

**Comment by jamesmengo** [comment]
* **#5042** <a href="https://app.graphite.dev/github/pr/Shopify/cli/5042?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/Shopify/cli/5042?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/theme/src/cli/utilities/theme-fs.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ const THEME_DIRECTORY_PATTERNS = [
 
 const THEME_PARTITION_REGEX = {
   sectionLiquidRegex: /^sections\/.+\.liquid$/,
-  liquidRegex: /\.liquid$/,
```
[the remark this answers is not in the document]
Comment (jamesmengo) [reply packages/theme/src/cli/utilities/theme-fs.ts:37]: surprisingly not. Probably an artifact from before we had more specific expressions

**Review by frandiox — APPROVED** [review]
LGTM!

**Review by EvilGenius13 — APPROVED** [review]
Changes make sense. Thanks for the quick fix! 🚀

**Pull request merged**